features/smoke_tests/handled.feature

```diff
@@ -0,0 +1,164 @@
+Feature: Handled smoke tests
+
+Scenario: Notify caught Java exception with default configuration
+    When I run "HandledJavaSmokeScenario"
+    And I wait to receive a request
+    Then the request is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier
+
+    # Exception details
+    And the payload field "events" is an array with 1 elements
+    And the exception "errorClass" equals "java.lang.RuntimeException"
+    And the exception "message" equals "HandledJavaSmokeScenario"
+    And the exception "type" equals "android"
+    And the payload field "events.0.exceptions.0.stacktrace" is a non-empty array
+    And the event "unhandled" is false
+    And the event "severity" equals "warning"
+    And the event "severityReason.type" equals "handledException"
+
+    # App data
+    And the event "app.buildUUID" is not null
+    And the event "app.id" equals "com.bugsnag.android.mazerunner"
+    And the event "app.releaseStage" equals "production"
+    And the event "app.type" equals "android"
+    And the event "app.version" equals "1.1.14"
+    And the event "app.versionCode" equals 34
+    And the event "app.duration" is not null
+    And the event "app.durationInForeground" is not null
+    And the event "app.inForeground" is true
+    And the payload field "events.0.metaData.app.memoryUsage" is greater than 0
+    And the event "metaData.app.name" equals "MazeRunner"
+    And the event "metaData.app.lowMemory" is not null
+
+    # Device data
+    And the payload field "events.0.device.cpuAbi" is a non-empty array
+    And the event "device.jailbroken" is false
+    And the event "device.id" is not null
+    And the payload field "events.0.device.id" is stored as the value "device_id"
+    And the event "device.locale" is not null
+    And the event "device.manufacturer" is not null
+    And the event "device.model" is not null
+    And the event "device.osName" equals "android"
+    And the event "device.osVersion" is not null
+    And the event "device.runtimeVersions" is not null
+    And the event "device.runtimeVersions.androidApiLevel" is not null
+    And the event "device.runtimeVersions.osBuild" is not null
+    And the payload field "events.0.device.totalMemory" is greater than 0
+    And the payload field "events.0.device.freeDisk" is greater than 0
+    And the payload field "events.0.device.freeMemory" is greater than 0
```

Comment:
Is there any overall preference for `the event` vs `the payload field` as mazerunner steps? This probably doesn't require immediate action, but it'd be nice to get the naming consistent

Reply:
As with implementing type checks this will be spun out into a separate ticket.